### PR TITLE
Refactored code for runtime permissions

### PR DESCRIPTION
**Pull request by jtj9817**
## Description
Removed the runtime permissions being asked at the start of the app and instead selectively implemented it on the parts of the app where it would be needed.

Fixes #(issue)
Possible fix to #566 

## Type of change
Just put an x in the [] which are valid.
- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.
- [x] `./gradlew assembleDebug assembleRelease`
- [x] `./gradlew checkstyle`

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: Android/app/src/main/java/io/github/project_travel_mate/MainActivity.java [hunk 1]
```diff
@@ -141,9 +141,6 @@ protected void onCreate(Bundle savedInstanceState) {
 
         fragmentManager.beginTransaction().replace(R.id.inc, fragment).commit();
 
-        // Get runtime permissions for Android M
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/MainActivity.java]: why did you remove these? I didnt get the point for this

Comment (jtj9817) [reply Android/app/src/main/java/io/github/project_travel_mate/MainActivity.java]: I removed these permissions because they will trigger the permissions to be asked if the user is just going to log-in or sign-up. I wanted the permissions to be only used at activities/app sections where it is only necessary. 

@thedevelopersanjeev 's implementation could be better but it's already asking for permissions at first time use for user and it doesn't take into account if the user declines all permissions.

File: Android/app/src/main/java/io/github/project_travel_mate/travel/HotelsActivity.java [hunk 2]
```diff
@@ -111,6 +115,25 @@ protected void onCreate(Bundle savedInstanceState) {
         }
     }
 
+    //Get permissions for making calls and locations
+    private void getPermissions() {
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/travel/HotelsActivity.java]: we would need runtime permissions for other places too (not just hotels?)

**Comment by prabhakar267** [comment]
@jtj9817 Any updates?

**Comment by prabhakar267** [comment]
Closing this due to inactivity.
